python/hail/api2/matrixtable.py

```diff
@@ -2036,32 +2036,36 @@ def filter_cols_list(self, keys, keep=True):
     def reorder_columns(self, mapping):
         """Reorder columns.
 
+        .. include:: ../_templates/req_tstring.rst
+
         Examples
         --------
 
-        Randomly shuffle order of samples:
+        Randomly shuffle order of columns:
 
         >>> import random
-        >>> new_sample_order = vds.sample_ids[:]
+        >>> new_sample_order = [x.s for x in dataset.cols_table().select("s").collect()]
         >>> random.shuffle(new_sample_order)
-        >>> vds_reordered = vds.reorder_samples(new_sample_order)
+        >>> vds_reordered = dataset.reorder_columns(new_sample_order)
 
         Notes
         -----
 
-        This method requires unique sample ids. `mapping` must contain the
-        same ids as :meth:`~hail.VariantDataset.sample_ids`. The order of the
-        ids in `mapping` determines the sample id order in the output dataset.
+        This method requires the keys to be unique. `mapping` must contain the
```

Comment:
I think it's clearer to rename the parameter `order`, since it directly lists the column keys in their desired order without reference to the old id order (indeed, look at the notes and parameter description calling it an order). `mapping` sounds like the parameter maps from column_ids to column_ids (map of strings) or old column order to new column order (map of indices).